android-sdk/src/main/java/org/sagebionetworks/bridge/android/data/StudyUploadEncryptor.java

```diff
@@ -0,0 +1,63 @@
+package org.sagebionetworks.bridge.android.data;
+
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.spongycastle.cms.CMSAlgorithm;
+import org.spongycastle.cms.CMSEnvelopedDataStreamGenerator;
+import org.spongycastle.cms.CMSException;
+import org.spongycastle.cms.jcajce.JceCMSContentEncryptorBuilder;
+import org.spongycastle.cms.jcajce.JceKeyTransRecipientInfoGenerator;
+import org.spongycastle.operator.OutputEncryptor;
+
+import java.io.FilterOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.security.cert.CertificateEncodingException;
+import java.security.cert.X509Certificate;
+
+import static com.google.common.base.Preconditions.checkState;
+
+/**
+ * Encrypts data using a Study's public key, so Bridge can decrypt it after upload.
+ */
+public class StudyUploadEncryptor {
+    private static final Logger LOG = LoggerFactory.getLogger(StudyUploadEncryptor.class);
+
+    private final Supplier<JceKeyTransRecipientInfoGenerator> recipientInfoGeneratorSupplier;
+
+    public StudyUploadEncryptor(X509Certificate publicKey) {
+        this.recipientInfoGeneratorSupplier = Suppliers.memoize(() -> {
+            try {
+                return new JceKeyTransRecipientInfoGenerator(publicKey).setProvider("SC");
```

Comment:
Spongy Castle, the encryption provider we're depending on. (BouncyCastle patched for Android)
I'll document this and make this a constant